johnzon-mapper/src/main/java/org/apache/johnzon/mapper/Mappings.java

```diff
@@ -332,39 +329,50 @@ public static boolean isPrimitive(final Type type) {
         return false;
     }
 
-    public ClassMapping getClassMapping(final Type clazz) {
+    public TypeMapping getClassMapping(final Type clazz) {
         return classes.get(clazz);
     }
 
-    public ClassMapping findOrCreateClassMapping(final Type clazz) {
-        ClassMapping classMapping = classes.get(clazz);
+    public TypeMapping findOrCreateTypeMapping(final Type clazz) {
+        return findOrCreateTypeMapping(clazz, false);
+    }
+
+    public TypeMapping findOrCreateTypeMapping(final Type clazz, final boolean applyObjectConverter) {
+        TypeMapping classMapping = classes.get(clazz);
         if (classMapping == null) {
-            if (!Class.class.isInstance(clazz) || Map.class.isAssignableFrom(Class.class.cast(clazz))) {
-                return null;
+            if (Class.class.isInstance(clazz)) {
+                if (!Map.class.isAssignableFrom(Class.class.cast(clazz))){
+                    classMapping = createTypeMapping(Class.class.cast(clazz), applyObjectConverter);
+                }
+            } else if (ParameterizedType.class.isInstance(clazz)){
+                classMapping = findOrCreateTypeMapping(ParameterizedType.class.cast(clazz).getRawType(), applyObjectConverter);
```

Comment:
I'd add the Stream.of(pt.getActualParameters()).map(it -> findOrCreateTypeMapping(it).collect(toList()) in the model + probably the CollectionMapping meta here to "normalize" the whole model and make it consistent

side note: take care for types which look like Foo<? extends Foo> and avoid loops (some tests to add pby)

Reply:
What model would you add the list to? Do I have to restore ParemeterizedTypeMapping?